src/services/clients/cxClient.ts

```diff
@@ -156,6 +159,49 @@ export class CxClient {
         return scanResult;
     }
 
+    async updateProjectCustomFields(): Promise<void> {
+        this.log.info("Updating Project Custom Fields.");
+        let projectId = this.projectId;
+        let path = `projects/${projectId}`;
+
+        //reading project custom fields entered by user on UI.
+        const projectCustomFields = this.sastConfig.projectCustomFields.split(",");
+        const projectCustomFieldsIds = new Array<string>(projectCustomFields.length);
+        const projectCustomFieldsKeys = new Array<string>(projectCustomFields.length);
+        const projectCustomFieldsValues = new Array<string>(projectCustomFields.length);
+        for (let i = 0; i < projectCustomFields.length; i++) {
+            projectCustomFieldsKeys[i] = projectCustomFields[i].split(":")[0];
+            projectCustomFieldsValues[i] = projectCustomFields[i].split(":")[1];
+        }
+
+        //reading project custom fields stored in SAST Portal
+        const fetchSASTProjectCustomFields = await this.httpClient.getRequest('customFields',{});
+        for (let i = 0; i < projectCustomFieldsKeys.length; i++){
+            for(let fetchSASTProjectCustomField of fetchSASTProjectCustomFields){
+                if(projectCustomFieldsKeys[i] === fetchSASTProjectCustomField.name){
```

Comment:
What if projectCustomFieldsKeys[i] has leading or trailing space characters

Reply:
It does not have it. This is the payload response from server so it has defined structure.